File: 프로그래머스/0/181871. 문자열이 몇 번 등장하는지 세기/문자열이 몇 번 등장하는지 세기.js
function countString(myString, pat) {
    let count = 0;
    let check = true;
    
    for (let i = 0; i < myString.length; i++) {
        for (let j = 0; j < pat.length; j++) {
            if (myString[i+j] !== pat[j]) {
                check = false;
                break;
            }
        }
        if (check) count++;
        check = true;
    }
    
    return count;
}

function solution(myString, pat) {
    let answer = 0;
    
    answer = countString(myString, pat);
    
    return answer;
}

File: 프로그래머스/0/181871. 문자열이 몇 번 등장하는지 세기/문자열이 몇 번 등장하는지 세기.java
class Solution {
    public int solution(String myString, String pat) {
        int answer = 0;
        int idx = -1;
        int prev = 0;

        while (myString.indexOf(pat, idx + 1) != -1) {
            if (idx == prev) {
                idx = myString.indexOf(pat, idx + 1);
                continue;
            }
            answer++;
            prev = idx;
        }

        return answer;
    }
}